src/pybind/mgr/balancer/module.py

```diff
@@ -377,25 +377,46 @@ def shutdown(self):
         self.run = False
         self.event.set()
 
-    def time_in_interval(self, tod, begin, end):
-        if begin <= end:
-            return tod >= begin and tod < end
+    def time_permit(self):
+        local_time = time.localtime()
+        time_of_day = time.strftime('%H%M', local_time)
+        weekday = (local_time.tm_wday + 1) % 7 # be compatible with C
+        permit = False
+
+        begin_time = self.get_config('begin_time') or '0000'
+        end_time = self.get_config('end_time') or '2400'
+        if begin_time <= end_time:
+            permit = begin_time <= time_of_day < end_time
         else:
-            return tod >= begin or tod < end
+            permit = time_of_day >= begin_time or time_of_day < end_time
+        if not permit:
+            self.log.debug("should run between %s - %s, now %s, skipping",
+                           begin_time, end_time, time_of_day)
+            return False
+
+        begin_weekday = int(self.get_config('begin_weekday')) or 0
```

Comment:
@k0ste 
See a lot of errors:
```
 7fcfc96a9700 -1 balancer.serve:
2019-02-22T13:33:35.456 INFO:tasks.ceph.mgr.x.smithi098.stderr:2019-02-22 13:33:35.453750 7fcfc96a9700 -1 Traceback (most recent call last):
2019-02-22T13:33:35.456 INFO:tasks.ceph.mgr.x.smithi098.stderr:  File "/usr/lib/ceph/mgr/balancer/module.py", line 419, in serve
2019-02-22T13:33:35.456 INFO:tasks.ceph.mgr.x.smithi098.stderr:    if self.active and self.time_permit():
2019-02-22T13:33:35.456 INFO:tasks.ceph.mgr.x.smithi098.stderr:  File "/usr/lib/ceph/mgr/balancer/module.py", line 397, in time_permit
2019-02-22T13:33:35.456 INFO:tasks.ceph.mgr.x.smithi098.stderr:    begin_weekday = int(self.get_config('begin_weekday')) or 0
2019-02-22T13:33:35.457 INFO:tasks.ceph.mgr.x.smithi098.stderr:TypeError: int() argument must be a string or a number, not 'NoneType'

```

Should be:
```python
begin_weekday = int(self.get_config('begin_weekday', 0))
end_weekday = int(self.get_config('end_weekday', 7))

Reply:
Updated